### PR TITLE
 feat: Implement test for POST resource

### LINKED ISSUE #646: Implement testcase for POST device resource
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]
This feature request is for metadata

### Description [**REQUIRED**]
According to https://docs.edgexfoundry.org/2.2/design/adr/core/0021-Device-Profile-Changes/ & https://github.com/edgexfoundry/edgex-go/issues/3925
The POST device resource implement tests would be added to metadata.

### Describe the solution you'd like
 If you have a solution in mind, please describe it.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by jumpingliu**
fix #646

### CODE REVIEW

File: TAF/testData/core-metadata/resource-profile.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (jinlinGuan) [inline TAF/testData/core-metadata/resource-profile.json:1]: This is not a valid json file which contains duplicate `resource` properties. Please refer to the json files in the `testData`

File: TAF/utils/scripts/docker/run-tests.sh [hunk 2]
```diff
@@ -100,4 +100,4 @@ if [ "$DEPLOY_SERVICES" != "no-deployment" ]; then
           -v /var/run/docker.sock:/var/run/docker.sock ${TAF_COMMON_IMAGE} \
           --exclude Skipped --include shutdown-edgex -u shutdown.robot -p default
   cp ${WORK_DIR}/TAF/testArtifacts/reports/edgex/log.html ${WORK_DIR}/TAF/testArtifacts/reports/cp-edgex/shutdown.html
-fi
+fi
```
Comment (jinlinGuan) [inline TAF/utils/scripts/docker/run-tests.sh:103]: This change is unnecessary.

File: TAF/testCaseModules/keywords/common/commonKeywords.robot [hunk 3]
```diff
@@ -203,6 +203,11 @@ Load yaml file "${yaml_file}" and convert to dictionary
     ${dict}=  yaml.Safe Load  ${yaml_data}
     [Return]  ${dict}
 
+Load json file "${json_file}" and convert to dictionary
```
Comment (jinlinGuan) [inline TAF/testCaseModules/keywords/common/commonKeywords.robot:206]: A similar keyword already exists. 
https://github.com/edgexfoundry/edgex-taf/blob/279746c4e7ace1f145c2d5a5232eb71ca2f1b5d4/TAF/testCaseModules/keywords/common/commonKeywords.robot#L196-L199

File: TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot [hunk 4]
```diff
@@ -438,6 +439,42 @@ Delete Profile Files
...
+    [Arguments]  @{data_list}
+    ${profile_list}=  Create List
+    FOR  ${json}  IN  @{data_list}
+        Set to dictionary  ${json}       apiVersion=${API_VERSION}
+        Append To List  ${profile_list}  ${json}
+    END
+    Set Test Variable  ${resourceProfile}  ${profile_list}
+
+Generate multiple resources
+     [Arguments]  ${resource_profile_1}  ${resource_profile_1}  ${resource_profile_1}
```
Comment (jinlinGuan) [inline TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot:464]: These arguments are not used in this keyword.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 5]
```diff
@@ -4,84 +4,117 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
...
 
 *** Test Cases ***
 ErrProfileResourcePOST001 - Add deviceResource with Empty device profile name
     # Empty device profile name
-    When Add New Resource With Empty device profile name
-    Then Should Return Status Code "404"
+    Given Set Test Variable  ${test_profile}  Test-Profile-1
+    And Generate a device profile sample  ${test_profile}
+    And Create Device Profile ${deviceProfile}
+    And Add multiple Resources on device profile
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:18]: Those can be moved to `Add multiple Resources on device profile` keyword for reuse.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot [hunk 6]
```diff
@@ -4,44 +4,59 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
...
-    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And New resources Should Be Added
-    [Teardown]  Delete Multiple Device Profiles By Names
+    And New Resource SwitchButton in Test-Profile-1 Should Be Added
+    And New Resource Temperature in Test-Profile-2 Should Be Added
+    And New Resource BinaryInput1 in Test-Profile-3 Should Be Added
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
+*** Keywords ***
+New Resource ${resource_name} in ${profile_name} Should Be Added
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:44]: Only check the new resource name exists in the resource list.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot [hunk 7]
```diff
@@ -4,84 +4,117 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
 Suite Setup  Run Keywords  Setup Suite
 ...                        AND  Run Keyword if  $SECURITY_SERVICE_NEEDED == 'true'  Get Token
 Suite Teardown  Run Teardown Keywords
-Force Tags      Skipped
-
+Force Tags      v2-api
 *** Variables ***
 ${SUITE}          Core Metadata Device Profile POST Deviceresource Negative Test Cases
+${profileName}    profileName
```
Comment (cherrycl) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot:10]: The variable is not used.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot (hunk 5 above)
Comment (jumpingliu) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot]: updated

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot (hunk 6 above)
Comment (jumpingliu) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Positive-Deviceresource.robot:41]: updated

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot (hunk 7 above)
Comment (jumpingliu) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/POST-Negative-Deviceresource.robot]: updated

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**